addons/product/models/product.py

```diff
@@ -683,11 +683,17 @@ def _is_variant_possible(self, parent_combination=None):
         return self.product_tmpl_id._is_combination_possible(self.product_template_attribute_value_ids, parent_combination=parent_combination, ignore_no_variant=True)
 
     def toggle_active(self):
-        """ Archiving related product.template if there is only one active product.product """
-        with_one_active = self.filtered(lambda product: len(product.product_tmpl_id.with_context(active_test=False).product_variant_ids) == 1)
-        for product in with_one_active:
-            product.product_tmpl_id.toggle_active()
-        return super(ProductProduct, self - with_one_active).toggle_active()
+        """ Archiving related product.template if there is not any more active product.product
+        (and vice versa, unarchiving the related product template if there is now an active product.product) """
+        result = super().toggle_active()
+        to_deactivate = self.filtered(lambda product: (not product.active
+                                                       and product.product_tmpl_id.active
+                                                       and len(product.product_tmpl_id.product_variant_ids) == 0)).mapped('product_tmpl_id')
+        to_activate = self.filtered(lambda product: (product.active
+                                                     and not product.product_tmpl_id.active)).mapped('product_tmpl_id')
+        to_deactivate.toggle_active()
+        to_activate.toggle_active()
```

Comment:
It works in one line as well:
```suggestion
        (tmpl_to_deactivate | tmpl_to_activate).toggle_active()
```

Reply:
Is there one that is worse for performance?  If there are 2 potentially large recordsets to combine?

Reply:
I don't think there should be a huge difference. Combining 2 recordsets is fast. But indeed, that may be an unnecessary operation.

Reply:
I'm going to ask the expert :)  but I do like it better on one line

Reply:
So the consensus was that the perf impact of combining the recordsets would be negligible and dwarfed by the write in any case.  There could be a case for using `write` instead of `toggle_active`, as in `to_activate.write({'active': True})`, because `toggle_active` won't batch the write, but it's also not a big perf problem as of v13 because the ORM will still batch them behind the scene.